PROCESS.md

```diff
@@ -0,0 +1,72 @@
+## STAC Development & Release Process
```

Comment:
I'm wondering whether we should have a central place for things that are common across stac-spec and stac-api-spec. It's always a pain to keep things aligned...

Reply:
I'm open to it, but not sure how it would work. I feel like it either is harder for a new user to understand (go look at this repo to find out x, y & z), or it's some complicated technical thing (though maybe submodules come close)?

I sorta figured the release processes would also diverge some.